projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,313 @@
+package projects.opentrain.gtfs_pipeline.heatmap;
+
+import com.projects.opentrain.gtfs_pipeline.Protos.Stop;
+import com.projects.opentrain.gtfs_pipeline.Protos.StopTime;
+import com.projects.opentrain.gtfs_pipeline.Protos.Trip;
+import com.google.startupos.common.flags.Flags;
+
+import javax.imageio.ImageIO;
+
+import java.awt.Graphics2D;
+import java.awt.Color;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.startupos.common.flags.Flag;
+import com.google.startupos.common.flags.FlagDesc;
+
+public class HeatMap {
+
+  public List<Stops_around> stopsAround = new ArrayList<>();
+
+  public int[][] heatMap;
+
+  @FlagDesc(
+      name = "image_path",
+      description = "Image path to create heatmap image,need to contain .../imageName.png")
+  public static Flag<String> imagePath = Flag.create("");
+
+  public class Stops_around // the stop and his location on arrival_time matrix
+   {
+
+    int i = 0; // index on heatMap
+    int j = 0;
+    Stop stop;
+    double arrivalTime;
+
+    public Stops_around(StopTime stopT, double arrivalT) {
+
+      arrivalTime = arrivalT;
+      Stop s;
+      FileInputStream StopInput = null;
+
+      try {
+        StopInput =
+            new FileInputStream(new File("projects/opentrain/gtfs_pipeline/stops.protobin"));
+      } catch (FileNotFoundException e) {
+        e.printStackTrace();
+      }
+
+      while (true) { // from all stops class pick my stop from stopsTime *to use lon/lat
```

Comment:
I don't think so because i choose stop from stopTime(i need DepartureTime and Stopsequence) and on this class i need stop from Stops class(i need Longitude and Latitude).

Reply:
and on other methods,
I think that it's the best way to read the objects in the file until there is no more objects and there can be some comments and things that I don't need.